packages/driver/src/cypress/stack_utils.ts

```diff
@@ -60,6 +60,55 @@ const stackWithLinesRemoved = (stack, cb) => {
   return unsplitStack(messageLines, remainingStackLines)
 }
 
+const stackTrimmedToTestInvocation = (stack: string, specWindow) => {
+  const modifiedStack = stackWithLinesRemoved(stack, (lines: string[]) => {
+    // Guard against Cypress being undefined/null (can happen when users quickly reload tests)
+    if (!specWindow?.Cypress) {
+      return lines
+    }
+
+    if (specWindow.Cypress.isBrowser({ family: 'chromium' })) {
+    // There are cases where there are other lines in the stack trace before the invocation (eg. `context.it.only`, `createRunnable`, etc)
+    // The actual test invocation line starts with either 'at eval' or 'at Suite.eval',
+    // so remove all lines until we reach the test invocation line
+      return _.dropWhile(lines, (line) => {
+        return !(
+          line.trim().startsWith('at eval') ||
+          line.trim().startsWith('at Suite.eval')
+        )
+      })
+    }
+
+    if (specWindow.Cypress.isBrowser({ family: 'firefox' })) {
+      const isTestInvocationLine = (line: string) => {
+        const splitAtAt = line.split('@')
+
+        // firefox stacks traces look like:
+        // functionName@http://localhost:3000/__cypress/tests?p=cypress/support/e2e.js:444:14
+        // @http://localhost:3000/__cypress/tests?p=cypress/e2e/spec.cy.js:43:3
+        // @http://localhost:3000/__cypress/tests?p=cypress/e2e/spec.cy.js:45:12
+        // evalScripts/<@cypress:///../driver/src/cypress/script_utils.ts:38:23
+        //
+        // the actual invocation details will be at the first line with no function name
+        return splitAtAt.length > 1 && splitAtAt[0].trim().length === 0
+      }
+
+      return _.dropWhile(lines, (line) => {
+        return !isTestInvocationLine(line)
+      })
+    }
+
+    return lines
+  })
+
+  // if we removed all the lines then something went wrong. return the original stack instead
+  if (modifiedStack.trim() === 'Error') {
```

Comment:
This seems brittle, I think in the `stackWithLinesRemoved` callback above, you can create a copy of the `lines` and then do your processing on that copy. Before returning `lines`, you can check if all of the lines were removed and if they were, you can return the original `lines`, otherwise return the copy.